### ISSUE TITLE
NPE when activating a mothballed mech

### ISSUE THREAD

**Issue by repligator** [MegaMek/mekhq#4658]
### Environment
Linux
Java 17
Master (Latest PR https://github.com/MegaMek/mekhq/commit/b3e0e4457a766489e5545398116ef0d341862003)

### Description
I attempted to activate a mothballed mech, and an NPE was thrown. The autosave is from before the activation, the manual save is from after
```
mekhq.MekHQ.lambda$main$1(MekHQ.java:294) - Uncaught Exception Detected
java.lang.NullPointerException: Cannot invoke "mekhq.campaign.personnel.Person.getId()" because the return value of "mekhq.campaign.unit.Unit.getCommander()" is null
	at mekhq.campaign.force.Force.lambda$getEligibleCommanders$0(Force.java:438)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
```

### Files
[mekhq.log](https://github.com/user-attachments/files/16637732/mekhq.log)
[Autosave-1-Yakuza38-30391110.cpnx.gz](https://github.com/user-attachments/files/16637741/Autosave-1-Yakuza38-30391110.cpnx.gz)
[NPE-Yakuza3830391111.cpnx.gz](https://github.com/user-attachments/files/16637745/NPE-Yakuza3830391111.cpnx.gz)

**Comment by IllianiCBT** [comment]
This is one of mine, I’m away for the weekend, but will get onto it on Sunday/Monday

**Comment by NickAragua** [comment]
This is likely fixed by #4671

**Comment by IllianiCBT** [comment]
Confirmed as resolved by Nick's PR. Closing as resolved.
